test-specifications/crypto-service/accountDeleteTransaction.md

```diff
@@ -90,12 +94,11 @@ The tests contained in this specification will assume that a valid account was a
-|---------|--------------------------------------------------------------------------|---------------------------------------------------------------------------------------------------------------------------------------------------|------------------------------------------------------------------------------------------------------------|-------------------|
-| 1       | Deletes an account with a transfer account                               | deleteAccountId=<CREATED_ACCOUNT_ID>, transferAccountId=<OPERATOR_ACCOUNT_ID>, commonTransactionParams.signers=[<PRIVATE_KEY_OF_CREATED_ACCOUNT>] | The account deletion succeeds.                                                                             | N                 |
-| 2       | Deletes an account without signing with the account's private key        | deleteAccountId=<CREATED_ACCOUNT_ID>, transferAccountId=<OPERATOR_ACCOUNT_ID>                                                                     | The account deletion fails with an INVALID_SIGNATURE response code from the network.                       | N                 |
-| 3       | Deletes an account with a transfer account that is the deleted account   | deleteAccountId=<CREATED_ACCOUNT_ID>, transferAccountId=<CREATED_ACCOUNT_ID>, commonTransactionParams.signers=[<PRIVATE_KEY_OF_CREATED_ACCOUNT>]  | The account deletion fails with an TRANSFER_ACCOUNT_SAME_AS_DELETE_ACCOUNT response code from the network. | N                 |
-| 4       | Deletes an account with a transfer account that is invalid/doesn't exist | deleteAccountId=<CREATED_ACCOUNT_ID>, transferAccountId="1000000.0.0", commonTransactionParams.signers=[<PRIVATE_KEY_OF_CREATED_ACCOUNT>]         | The account deletion fails with an INVALID_TRANSFER_ACCOUNT_ID response code from the network.             | N                 |
+| Test no | Name                                                                     | Input                                                                                                                                            | Expected response                                                                                          | Implemented (Y/N) |
+|---------|--------------------------------------------------------------------------|--------------------------------------------------------------------------------------------------------------------------------------------------|------------------------------------------------------------------------------------------------------------|-------------------|
+| 1       | Deletes an account with no transfer account                              | deleteAccountId=<CREATED_ACCOUNT_ID>, commonTransactionParams.signers=[<PRIVATE_KEY_OF_CREATED_ACCOUNT>]                                         | The account deletion fails with an ACCOUNT_ID_DOES_NOT_EXIST response code from the network.               | N                 |
```

Comment:
Not sure how to achieve it as transfer account is optional.

Reply:
A transfer account it seems is required by the network. I believe not filling in a value for the `transferAccountId` in the protobuf will default it to a `0.0.0` value, hence the `ACCOUNT_ID_DOES_NOT_EXIST` status, but not 100% sure on that

Reply:
Got it. I can see it is optional in Java SDK. Can you please check it more in C++ SDK?

Reply:
Yes, I can confirm it leaves `transferAccountId` with a `0.0.0` value before and after serialization/deserialization

Reply:
Understood, then we have difference in this between SDKs 🤔 
cc @agadzhalov @0xivanov (fyi and can you pls check JS and Go?)

Reply:
Just for clarity, C++ SDK also has `transferAccountId` as optional in the `AccountDeleteTransaction` class like the other SDKs. However, it seems that when the transaction gets built into a `CryptoDeleteTransactionBody` protobuf, if the `transferAccountId` doesn't have a value, the value in the protobuf defaults to the `0.0.0` value. It also doesn't fill out the `account_case`, which is just an enum that defines in the protobuf if the third value is an account number or an alias. So the consensus node could also be reading this `account_case` as the `ACCOUNT_NOT_SET` enum and perhaps giving an unclear response with the `ACCOUNT_ID_DOES_NOT_EXIST` instead of something like `INVALID_ACCOUNT_ID`.

Reply:
Thanks for checking and clarifications 👍🏻 
I think main question here is how to achieve this case through SDK API 🤔 is it possible in C++ SDK?

Reply:
Running
```
AccountDeleteTransaction()
                .setDeleteAccountId(newAccountId)
                .freezeWith(&client)
                .sign(privateKey)
                .execute(client)
                .getReceipt(client);
```

gives me
`Hedera::PrecheckStatusException: Hedera transaction 0.0.3661843@1721755051.68870000 failed precheck with status ACCOUNT_ID_DOES_NOT_EXIST`

Reply:
I am sorry, I was confused (bc was just looking at the code). [Docs page](https://docs.hedera.com/hedera/sdks-and-apis/sdks/accounts-and-hbar/delete-an-account) say it is **required** and I wrote example in Java SDK to confirm such behaviour 👍🏻